pkg/apply/prune/prune.go

```diff
@@ -150,7 +150,11 @@ func (po *PruneOptions) Prune(localInv inventory.InventoryInfo,
 		// Handle lifecycle directive preventing deletion.
 		if !canPrune(localInv, obj, o.InventoryPolicy, uid) {
 			klog.V(4).Infof("skip prune for lifecycle directive %s/%s", pruneObj.Namespace, pruneObj.Name)
-			taskContext.EventChannel() <- createPruneEvent(pruneObj, obj, event.PruneSkipped)
+			e := createPruneEvent(pruneObj, obj, event.PruneSkipped)
+			if po.Destroy {
+				e = createDeleteEvent(pruneObj, obj, event.DeleteSkipped)
```

Comment:
This feels a little bit awkward, but we can see if there are ways to improve this later.

Reply:
It is a bit awkward, but it allows us to get rid of even more awkwardness with the channel transformers. Adding a TODO to improve this in the future.

Reply:
I created an `EventFactory` interface to abstract the prune/delete event generation.